UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Utils/AtmosUtils.cs

```diff
@@ -54,11 +55,12 @@ public static bool IsPressureChanged(MetaDataNode node, out Vector2Int windDirec
 					}
 					else
 					{
+
 						// check if the moles are different. (e.g. CO2 is different from breathing)
-						for (int j = 0; j < Gas.Count; j++)
+						foreach (var gas in Gas.Gases)
 						{
-							float moles = node.GasMix.Gases[j];
-							float molesNeighbor = neighbor.GasMix.Gases[j];
+							float moles = node.GasMix.GasData.GetGasMoles(gas.Key);
+							float molesNeighbor = neighbor.GasMix.GasData.GetGasMoles(gas.Key);
 
 							if (Mathf.Abs(moles - molesNeighbor) > AtmosConstants.MinPressureDifference)
 							{
```

Comment:
Currently this is looping through all possible gases, but it might be better if i changed it to loop through all current node, do the checks and then loop through all neighbour and do the checks there too. Could make a list containing the unique gas types between the two but that would GC

Reply:
Also just realised for an optimisation it should break after the setting to true when they are different on line 67 as that only needs to happen once.